File: didatico/julia/codes/randomsearch.f90
program randomsearch
  implicit none
  integer :: i, ntrial
  double precision :: random, x(2), f, fbest, xbest(2)
  ! Test 10000 points
  ntrial = 10000
  fbest = 1.d30
  do i = 1, ntrial
    call random_number(random)
    x(1) = -10.d0 + 20.d0*random
    call random_number(random)
    x(2) = -10.d0 + 20.d0*random
    call computef(x,f)
    if ( f < fbest ) then
      fbest = f
      xbest(1) = x(1)
      xbest(2) = x(2)
      write(*,*) i, ' New best point: ', x(1), x(2), ' f = ', f
    end if
  end do
  write(*,*) ' Best point found: ', xbest(1), xbest(2), ' f = ', fbest
end program randomsearch
!
! Subroutine that computes the function value
!
subroutine computef(x,f)
  implicit none
  double precision :: x(2), f
  f = x(1)**2 + x(2)**2
end subroutine computef

File: main/didatico/julia/codes/randomsearch.jl
#
# function that computes the function value
#
function computef(x)
  f = x[1]^2 + x[2]^2
  return f
end
#
# Program randomsearch
#
let RandomSearch
  # Test 10000 points
  x = [ 0. , 0. ]
  xbest = x
  ntrial = 10000
  fbest = computef(x)
  for i in 1:ntrial
    x[1] = -10.e0 + 20.e0*rand(Float64)
    x[2] = -10.e0 + 20.e0*rand(Float64)
    f = computef(x)
    if f < fbest 
      fbest = f
      xbest = x
      println( i, " New best point: ", x, " f = ", f )
    end
  end
  println(" Best point found: ", xbest, " f = ", fbest)
end
